code/modules/mob/living/carbon/damage_procs.dm

```diff
@@ -74,22 +76,29 @@
 /mob/living/carbon/adjustFireLoss(amount, updating_health = TRUE, forced = FALSE, required_status)
 	if(!forced && (status_flags & GODMODE))
 		return FALSE
+	if(amount == 0)
+		return FALSE
 	if(amount > 0)
 		take_overall_damage(0, amount, 0, updating_health, required_status)
 	else
 		heal_overall_damage(0, abs(amount), 0, required_status ? required_status : BODYPART_ORGANIC, updating_health)
 	return amount
 
 /mob/living/carbon/adjustToxLoss(amount, updating_health = TRUE, forced = FALSE)
-	if(!forced && HAS_TRAIT(src, TRAIT_TOXINLOVER)) //damage becomes healing and healing becomes damage
+	if(amount == 0)
+		return
+	var/tox_immune = HAS_TRAIT(src, TRAIT_TOXINLOVER)
+	if(!forced && tox_immune) //damage becomes healing and healing becomes damage
 		amount = -amount
-		if(amount > 0)
-			blood_volume -= 5*amount
-		else
-			blood_volume -= amount
 	if(HAS_TRAIT(src, TRAIT_TOXIMMUNE)) //Prevents toxin damage, but not healing
 		amount = min(amount, 0)
-	return ..()
+	. = ..()
+	if(isnull(.) || forced || !tox_immune)
+		return
+	if(amount > 0) //finish toxin lover damage if we actually changed health
+		blood_volume -= 5*amount
+	else
+		blood_volume -= amount
```

Comment:
This make imposible to create blood for slimepeople by eating toxins if slimeperson at max health?

Reply:
Is this still an issue?